packages/x-ui-collections-bundle/src/graphql/Collection.ts

```diff
@@ -749,4 +749,31 @@ export abstract class Collection<T = null> {
   protected isUpdateModifier(element: any): element is UpdateFilter<T> {
     return element["$set"];
   }
+
+  /**
+   * Integration with native `useQuery` from Apollo, offering a hybrid approach so apollo re-uses your cache.
+   */
+  public optimistic(_id: ObjectId | string, document: T) {
+    const stringId = _id instanceof ObjectId ? _id.toString() : _id;
+    const operationName = ""; // ! how do we determine this here ?
```

Comment:
based on this.getName(), we'll need to do it for Find FindOne and Count maybe? Or give some opts?

Reply:
The idea would be to have this behavior by default.

A developper will prefer to write :
```typescript
collection.updateOne(_id, document)
```
and expect the cache to be automatically update (or disable that behavior with a `optimisticReponse: false` option) than to have to manually add an additional for each update in his software :

```typescript
collection.updateOne(_id,
  document,
  { optimisticResponse: collection.optimistic(_id, document) }
)
```

We could add an additional optional argument with the `queryName`, that would default to `[Collection]UpdateOne`

Reply:
Proposition commited : [https://github.com/bluelibs/bluelibs/pull/183/commits/2a6aac5988fa991504a9cdeb72f5aad6a90085f9](https://github.com/bluelibs/bluelibs/pull/183/commits/2a6aac5988fa991504a9cdeb72f5aad6a90085f9)